### DISCUSSION TITLE
Multiple chromium instances and extension issues.

### DISCUSSION THREAD

**Discussion by dkebler** [keepassxreboot/keepassxc#10327] [Q&A] [answered]
Back in  the day when I was using ole keespass and the kee browser extension by davd lech
https://chromewebstore.google.com/detail/kee-password-manager/mmhlniccooihdimnnjhamobppdhaolme?hl=en&pli=1

I could have multiple databases open and my browser would find a matching entry from any open database.  I typically have 2 to 3 open.   In my experience with xc extension if I need a particular entry and that entry is in an open database that is not in foreground it is not found.   If I then switch to XC and bring that database to the front then going back to my browser and refresh the entry will be found.

So is there something that needs to be set for this to work like kee or is this not possible with extension (right now)?  

If not then can this be added to the extension code?   (i'll make an issue)

Not a show stopper but kinda annoying considering with kee and ole keepass I didn't have to do this.

**Comment by varjolintu** [comment]
KeePassXC Settings -> Browser integration -> [x] Search in all opened databases for matching credentials. Note that every database must be connected to the extension.

**Comment by dkebler** [comment]
well, that was already checked.  That's why I am asking

![image](https://github.com/keepassxreboot/keepassxc/assets/1316185/4976475b-7f05-4b98-be37-ca76e368b713)

**Comment by dkebler** [comment]
and yes both databases are connected as I said if I bring the other to foreground then credentials from that db will be found.  otherwise no.

**Reply by droidmonkey** [reply]
What version of keepassxc are you using?

**Comment by dkebler** [comment]
I think this is an issue with me running multiple chromium instances and thus "native-messaging-hosts" is unique path for each.  The advanced setting for broweser integration is for the entire app not per "connection"   Seems like in older versions I could set this  per connection but now I don't see this.  So maybe it is fixed?   I've removed all my "stored keys" and kinda "start from new".  

```
KeePassXC - Version 2.7.6
Revision: dd21def

Qt 5.15.12
Debugging mode is disabled.

Operating system: Manjaro Linux
CPU architecture: x86_64
Kernel: linux 6.1.71-1-MANJARO

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 3.2.0

```

**Reply by droidmonkey** [reply]
Somewhat important detail to include.

**Comment by dkebler** [comment]
extension keeps asking to connect like the connection info is not going into the right "NativeMessagingHosts" folder

**Comment by dkebler** [comment]
```
KeePassXC - 2.7.6
KeePassXC-Browser - 1.8.12
Operating system: Linux x86_64
Browser: Chrome/Chromium 120.0.0.0
```

**Comment by dkebler** [comment, marked as answer]
ok, copied the "new" json file `org.keepassxc.keepassxc_browser.json`  from ./config..  to each of the browser instance NativeMessangingHosts folder.    

that an reissuing the key for each database/browser instance and ...

It all seems to be working now as expected.     Looks like the uuid for the browser extension had changed now is 

        "chrome-extension://oboonakemofpalcgghocfoadofidjkkk/"
was 
        
        "chrome-extension://pdffhmdngciaglkoonimfcmckehcpafo/",


Is there no way that XC/extension can determine the actual folder for the chromium instance it's running in so to put that json file in the right place instead of just some default `~/.config/chromium/NativeMessagingHosts`


anyway it's fixed

**Reply by droidmonkey** [reply]
oboonakemofpalcgghocfoadofidjkkk is the extension from the Chrome store
pdffhmdngciaglkoonimfcmckehcpafo is the extension from the Edge store
